### PR TITLE
runtime-rs: Add a new config option for QEMU

### PR DESCRIPTION
**Pull request by BbolroC**
This PR deals with generating a configuration file `configuration-qemu.toml` if `QEMUCMD` is defined for an architecture of interest.

It also includes creating a symbolic link to `configuration-qemu.toml` if QEMU is defined as the default hypervisor.

A generated config file can you find at https://gist.github.com/BbolroC/48c60b5ab8d929a56cc3ef03d7da4245.

fixes: #8441

Signed-off-by: Hyounggyu Choi <Hyounggyu.Choi@ibm.com>

### CODE REVIEW

**Review by gkurz — COMMENTED** [review]
Hi @BbolroC ! Thanks for resurrecting this PR :-)

Looks mostly good. I've commented on some nits.

File: src/runtime-rs/Makefile [hunk 1]
```diff
@@ -245,14 +246,60 @@ ifneq (,$(CLHCMD))
...
+    DEFNETWORKMODEL_QEMU := tcfilter
+    DEFDISABLEGUESTSELINUX := true
+    DEFSECCOMPSANDBOXPARAM := on,obsolete=deny,spawn=deny,resourcecontrol=deny
+    DEFGUESTSELINUXLABEL := system_u:system_r:container_t
+endif
+
 ifeq ($(DEFAULT_HYPERVISOR),$(HYPERVISOR_DB))
     DEFAULT_HYPERVISOR_CONFIG = $(CONFIG_FILE_DB)
+else ifeq ($(DEFAULT_HYPERVISOR),$(HYPERVISOR_CLH))
+	DEFAULT_HYPERVISOR_CONFIG = $(CONFIG_FILE_CLH)
```
[the remark this answers is not in the document]
Comment (BbolroC) [reply src/runtime-rs/Makefile:291]: Okay, then I will rule it out from this PR.

File: src/runtime-rs/Makefile [hunk 2]
```diff
@@ -245,14 +246,60 @@ ifneq (,$(CLHCMD))
...
+else ifeq ($(DEFAULT_HYPERVISOR),$(HYPERVISOR_QEMU))
+	DEFAULT_HYPERVISOR_CONFIG = $(CONFIG_FILE_QEMU)
+else
+	$(error "ERROR: invalid default hypervisor: '$(DEFAULT_HYPERVISOR)'")
```
[the remark this answers is not in the document]
Comment (BbolroC) [reply src/runtime-rs/Makefile]: As discussed on Slack, I will remove this `else` arm and update the conditionals like those in Makefile for runtime. Thanks.

File: src/runtime-rs/config/configuration-qemu.toml.in [hunk 3]
```diff
@@ -1,5 +1,4 @@
-# Copyright (c) 2017-2019 Intel Corporation
-# Copyright (c) 2021 Adobe Inc.
+# Copyright (c) 2024 IBM Inc.
```
Comment (pmores) [inline src/runtime-rs/config/configuration-qemu.toml.in:1]: IANAL but shouldn't the previous copyrights be retained?

**Review by pmores — APPROVED** [review]
Thanks @BbolroC , looks good to me (though admittedly I couldn't properly verify every single change in the final commit).

File: src/runtime-rs/config/configuration-qemu.toml.in (hunk 3 above)
Comment (BbolroC) [reply src/runtime-rs/config/configuration-qemu.toml.in:3]: Yep, I made the original ones retained and simply added a new one. Thanks!

**Comment by BbolroC** [comment]
For reviewers, I've updated the PR by differentiating `DEFSANDBOXCGROUPONLY_QEMU` from `DEFSANDBOXCGROUPONLY`. It is because it overrides `DEFSANDBOXCGROUPONLY=true` for dragonball and cloud-hypervisor with `DEFSANDBOXCGROUPONLY=false` for qemu at the end  if `DBCMD`, `CLHCMD` and `QEMUCMD` is defined at the same time (if `ARCH` is x86_64 in this case). This leads to a test failure for draganball and cloud hypervisor like:


```
ailed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: Others("failed to handle message create container\n\nCaused by:\n
0: create\n    
1: using method in wrong cgroup mode."): unknown
```

**Pull request merged**